src/blocks/custom/form/assets/form.js

```diff
@@ -37,59 +54,158 @@ export class Form {
       success: window.eightshiftForms.content.formSuccess,
     };
 
-    this.STATE_IS_LOADING = false;
-    this.CLASS_FORM_SUBMITTING = 'form-submitting';
-    this.CLASS_HIDE_SPINNER = 'hide-spinner';
-    this.CLASS_HIDE_MESSAGE = 'hide-form-message';
-    this.CLASS_HIDE_OVERLAY = 'hide-form-overlay';
+    this.errorMessageClasses = [
+      'form-message',
+      'js-form-message',
+      'js-form-message--error',
+      'form-message__type--error',
+    ];
+
+    this.errors = [];
```

Comment:
![download](https://user-images.githubusercontent.com/8638515/96881264-5e921980-147e-11eb-9f76-ce9bbac15b49.png)

That's a big ass constructor 😬

You sure this class is not doing too much? 😅 

I mean if it works...

Reply:
Yeah it handles all logic for submitting forms, it is long I agree, and maybe I could refactor it to break it up into little pieces but I'd rather not do that right now 😅

Reply:
Yeah, if it's not broken don't fix it 😅 

But you have tests so it shouldn't be hard to refactor later on 💪